assets/js/base/hocs/with-reviews.js

```diff
@@ -13,14 +13,18 @@ import { formatError } from '../utils/errors.js';
 
 const withReviews = ( OriginalComponent ) => {
 	class WrappedComponent extends Component {
-		constructor() {
-			super( ...arguments );
+		constructor( props ) {
+			super( props );
+
+			this.isPreview = Boolean( props.attributes.previewReviews );
 
 			this.state = {
 				error: null,
 				loading: true,
-				reviews: [],
-				totalReviews: 0,
+				reviews: this.isPreview ? props.attributes.previewReviews : [],
+				totalReviews: this.isPreview
+					? props.attributes.previewReviews.length
+					: 0,
```

Comment:
would be a good opportunity here to refactor this to class properties and get rid of the constructor?

Reply:
Good idea! Done in 9ec372240baa707a384c6ac8bc400bfa8342d934.